articles/quasar-googleauth-with-firebase/index.md

```diff
@@ -0,0 +1,649 @@
+3. Disable Google Analytics for this project and click `Continue` to create our project.
+4. After the project is ready click `Continue`. It will take you to the project overview. Click the `web` icon, register an app name and name it `quasar-firebase-googleauth`, and click `Register App`.
+5. You'll be taken to `Step 2 - Add Firebase SDK` and copy the whole script.
+
+We're going to install firebase and create a boot file to initialize firebase before our app runs.
+
+Go back to our project, open the terminal, and run the below commands:
+
+```bash
+yarn add firebase
+quasar new boot firebase
+```
+
+Navigate to `quasar.conf.js` file and search for `boot`. It should be an empty array. Add the `firebase.js` boot file in it as indicated below.
+
+```JavaScript
+boot: ['firebase'],
+```
+
+Navigate to the `boot` folder and open the `firebase.js` file. 
+
+Paste the script you copied from the previous step for the firebase SDK. `firebase.js` file structure should resemble the one below.
+
+```JavaScript
+import firebase from "firebase";
+
+const firebaseConfig = {
+    apiKey: "",
+    authDomain: "",
+    projectId: "",
+    storageBucket: "",
+    messagingSenderId: "",
+    appId: ""
+  };
+  firebase.initializeApp(firebaseConfig);
+
+export default firebase
+```
+
+### Access Control for Routes
+
+We're going to disable access to routes that require authorization when a user needs to access them. This is quite easy. Navigate to the `routes` folder, open the `index.js` file.
+
+Add `import firebase from "firebase"` at the top of the file just like we did in the `firebase.js` boot file. When done, add the following code just above the line `return Router`:
+
+```JavaScript
+  Router.beforeEach(async (to, from, next) => {
+    const auth = to.meta.requiresAuth
+    if (auth && !await firebase.getCurrentUser()) {
+      next('/');
+    } else {
+      next();
+    }
+  })
+```
+
+Now if the currentUser is null or undefined, we should redirect users to the auth path (`/`). But how do we get currentUser? We can’t use `firebase.auth().currentUser` because on page refresh that property has not been set yet before the `requiresAuth` guard is triggered.
+
+We'll have to use the `onAuthStateChanged` callback somehow. We have to add a method to the firebase object after we initialize the firebase app. The method is added to the `firebase.js` boot file. Edit the firebase boot file to resemble the code below.
+
+```JavaScript
+const firebaseConfig = {
+    apiKey: "",
+    authDomain: "",
+    projectId: "",
+    storageBucket: "",
+    messagingSenderId: "",
+    appId: ""
+  };
+  firebase.initializeApp(firebaseConfig);
+
+  firebase.getCurrentUser = () => {
+    return new Promise((resolve, reject) => {
+      const unsubscribe = firebase.auth().onAuthStateChanged(user => {
+        unsubscribe();
+        resolve(user);
+      }, reject);
+    })
+  };
```

Comment:
Please format this code properly.

Reply:
this code is okay..I'm initializing firebase configuration when vue file boots up

Reply:
yeah, the code is okay. I just asked you to properly format the snippet. I did it for you this time.